Bot/PictureBot/Controllers/MessagesController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Description;
+using Microsoft.Bot.Connector;
+using Newtonsoft.Json;
+
+namespace PictureBot
+{
+    [BotAuthentication]
+    public class MessagesController : ApiController
+    {
+        /// <summary>
+        /// POST: api/Messages
+        /// Receive a message from a user and reply to it
+        /// </summary>
+        public async Task<HttpResponseMessage> Post([FromBody]Activity activity)
+        {
+            if (activity.Type == ActivityTypes.Message)
+            {
+                ConnectorClient connector = new ConnectorClient(new Uri(activity.ServiceUrl));
+                // calculate something for us to return
+                int length = (activity.Text ?? string.Empty).Length;
+
+                // return our reply to the user
+                Activity reply = activity.CreateReply($"You sent {activity.Text} which was {length} characters");
```

Comment:
Thanks Mike.  This is the verbatim Visual Studio template code when you create a new Bot Framework project.  I will be changing it later anyway since this is not an EchoBot.  I checked in the template code for 2 reasons: 1) to more easily see the differences from the template and 2) to have fresh clean versions of the Web.config file available before I add my specific bot IDs and such to it.